packages/dds/tree/src/test/simple-tree/treeApi.spec.ts

```diff
@@ -768,5 +768,147 @@ describe("treeNodeApi", () => {
+			);
+			const root = view.root;
+
+			let listenerFired = false;
+			Tree.on(root, "nodeChanged", ({ changedProperties }) => {
+				// This is the main validation for the test
+				assert.deepEqual(changedProperties, new Set(["key1", "key2", "key3"]));
+				listenerFired = true;
+			});
+
+			const { forkView, forkCheckout } = getViewForForkedBranch(view);
+
+			// The implementation details of the kinds of changes that can happen inside the tree are not exposed at this layer.
+			// But since we know then, try to cover all of them.
+			forkView.root.set("key1", 0); // Replace existing key
+			forkView.root.delete("key2"); // Remove a key
+			forkView.root.set("key3", 3); // Add new key
+
+			view.checkout.merge(forkCheckout);
+
+			// Validate changes actually took place and all listeners fired
+			assert.equal(root.get("key1"), 0, "'key1' value did not change as expected");
+			assert.equal(root.get("key2"), undefined, "'key2' value did not change as expected");
+			assert.equal(root.get("key3"), 3, "'key3' value did not change as expected");
+			assert.equal(listenerFired, true, "'nodeChanged' should have fired");
+		});
+
+		it(`'nodeChanged' includes the names of changed properties (arrayNode)`, () => {
+			const sb = new SchemaFactory("test");
+			class treeSchema extends sb.array("root", [sb.number]) {}
+
+			const view = getView(new TreeViewConfiguration({ schema: treeSchema }));
+			view.initialize([1, 2]);
+			const root = view.root;
+
+			let listenerFired = false;
+			Tree.on(root, "nodeChanged", ({ changedProperties }) => {
+				// This is the main validation for the test
+				assert.deepEqual(changedProperties, new Set());
+				listenerFired = true;
+			});
+
+			const { forkView, forkCheckout } = getViewForForkedBranch(view);
+
+			// The implementation details of the kinds of changes that can happen inside the tree are not exposed at this layer.
+			// But since we know then, try to cover all of them.
+			forkView.root.insertAtEnd(3); // Append to array
+			forkView.root.removeAt(0); // Remove from arrray
+			forkView.root.moveRangeToEnd(0, 1); // Move within array
+
+			view.checkout.merge(forkCheckout);
+
+			// Validate changes actually took place and all listeners fired
+			// Note: as of 2024-07-29, if the schema for the root is defined with a name
+			// (`class treeSchema extends sb.array("root", [sb.number])` where "root" is the name), we need to use Array.from()
+			// or array spread in order to compare the node to another array.
+			assert.deepEqual([...root], [3, 2]);
+			assert.equal(listenerFired, true, "'nodeChanged' should have fired");
+		});
+
+		it(`'nodeChanged' uses view keys, not stored keys, for the list of changed properties`, () => {
+			const sb = new SchemaFactory("test");
+			class treeSchema extends sb.object("root", {
+				prop1: sb.optional(sb.number, { key: "stored-prop1" }),
+				prop2: sb.optional(sb.number, { key: "stored-prop2" }),
+				prop3: sb.optional(sb.number, { key: "stored-prop3" }),
```

Comment:
quite a bit of this tests is redundant with others. I think it would be better to simplify this to just a single case, and one property so its more specifically testing the thing this test is about instead of replicating things like checking if `"'prop2' value did not change as expected"`

Reply:
Fair, particularly for this test. I'll keep one prop.